config/runtime.exs

```diff
@@ -0,0 +1,170 @@
+    ],
+    server_flag: "GB-WLS",
+    enable_benchmark: false,
+    node_name: Teiserver.ConfigHelpers.get_env("NODE_NAME"),
+    extra_logging: false,
+    enable_managed_lobbies: true,
+    user_agreement:
+      "A verification code has been sent to your email address. Please read our terms of service at https://#{domain_name}/privacy_policy and the code of conduct at https://www.beyondallreason.info/code-of-conduct. Then enter your six digit code below if you agree to the terms."
+
+  config :teiserver, Teiserver.Repo,
+    username: Teiserver.ConfigHelpers.get_env("DB_USERNAME"),
+    password: Teiserver.ConfigHelpers.get_env("DB_PASSWORD"),
+    database: Teiserver.ConfigHelpers.get_env("DB_NAME"),
+    pool_size: 40,
+    timeout: 120_000,
+    queue_interval: 2000
+
+  check_origin =
+    if Teiserver.ConfigHelpers.get_env("ACQUIRE_TLS_CERTIFICATE", false, :bool) do
+      ["//#{domain_name}", "//*.#{domain_name}"]
+    else
+      false
+    end
+
+  config :teiserver, TeiserverWeb.Endpoint,
+    url: [host: domain_name],
+    check_origin: check_origin,
+    https:
+      certificates ++
+        [
+          versions: [:"tlsv1.2"],
+          # dhfile is not supported for tls 1.3
+          # https://www.erlang.org/doc/man/ssl.html#type-dh_file
+          dhfile: Teiserver.ConfigHelpers.get_env("HTTPS_DH_FILE_PATH", "/etc/ssl/dhparam.pem")
+        ],
+    http: [:inet6, port: Teiserver.ConfigHelpers.get_env("PORT", "4000", :int)],
+    secret_key_base: Teiserver.ConfigHelpers.get_env("HTTP_SECRET_KEY_BASE")
+
+  config :teiserver, Teiserver.Account.Guardian,
+    issuer: Teiserver.ConfigHelpers.get_env("GUARDIAN_ISSUER", "teiserver"),
+    secret_key: Teiserver.ConfigHelpers.get_env("GUARDIAN_SECRET_KEY")
+
+  if Teiserver.ConfigHelpers.get_env("ENABLE_EMAIL_INTEGRATION", :bool) do
+    smtp_server = Teiserver.ConfigHelpers.get_env("SMTP_HOSTNAME")
+
+    config :teiserver, Teiserver.Mailer,
+      adapter: Bamboo.SMTPAdapter,
+      noreply_name: "Beyond All Reason",
+      noreply_address: "noreply@#{domain_name}",
+      contact_address: "info@#{domain_name}",
+      server: smtp_server,
+      hostname: smtp_server or Teiserver.ConfigHelpers.get_env("SMTP_SERVER"),
+      # port: 1025,
+      port: Teiserver.ConfigHelpers.get_env("SMTP_PORT", "587", :int),
+      username: Teiserver.ConfigHelpers.get_env("SMTP_USERNAME"),
+      password: Teiserver.ConfigHelpers.get_env("SMTP_PASSWORD"),
+      # tls: :if_available, # can be `:always` or `:never`
+      # can be `:always` or `:never`
+      tls: :always,
+      tls_verify:
+        if(Teiserver.ConfigHelpers.get_env("SMTP_TLS_VERIFY", true, :bool),
+          do: :verify_peer,
+          else: :verify_none
+        ),
+      # or {":system", ALLOWED_TLS_VERSIONS"} w/ comma seprated values (e.g. "tlsv1.1,tlsv1.2")
+      allowed_tls_versions: [:"tlsv1.2"],
+      # can be `true`
+      ssl: false,
+      retries: 1,
+      # can be `true`
+      no_mx_lookups: false,
```

Comment:
The 3 above variables are basically defaults, I would skip setting them entirely to reduce clutter.

Reply:
I've removed `ssl` and `retries`. It's not clear that `no_mx_lookups` defaults to false though so I left it.